src/netaddress.cpp

```diff
@@ -243,13 +244,15 @@ bool CNetAddr::IsTor() const { return m_net == NET_ONION; }
 bool CNetAddr::IsLocal() const
 {
     // IPv4 loopback (127.0.0.0/8 or 0.0.0.0/8)
-    if (IsIPv4() && (GetByte(3) == 127 || GetByte(3) == 0))
+    if (IsIPv4() && (m_addr[0] == 127 || m_addr[0] == 0)) {
         return true;
+    }
 
     // IPv6 loopback (::1/128)
     static const unsigned char pchLocal[16] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1};
-    if (IsIPv6() && memcmp(ip, pchLocal, 16) == 0)
+    if (IsIPv6() && memcmp(m_addr.data(), pchLocal, sizeof(pchLocal)) == 0) {
```

Comment:
If we for a brief moment pretended that `prevector.h:258` had this line added to it:
```C++
    prevector(std::initializer_list<T> il) : prevector(il.begin(), il.end()) {}
```

then it would have been worth it to do:
```C++
static const prevector<16, uint8_t> ipv6_loopback{0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1};
if (IsIPv6() && m_addr == ipv6_loopback) {
```

Alas, that prevector line would need to be added. Not sure it's worth it, or if it should be its own PR perhaps.

Reply:
I think it is also important to not deviate from `std::vector` so that it is not too much hassle should one decide to swap `prevector` for `std::vector` or the other way around.

Btw, `std::vector` has an initializer list constructor.

Anyway - leaving this as is.

Reply:
Yeah, it may be that my lines below (the `ipNode6`) were in fact invalid, and were the cause for the errors I saw when I did this as a prevector constructor, where I didn't initially cast 16 to `size_t`. I much prefer the constructor variant, which is also closer to `std::vector`. FTR, that was

```C++
prevector(std::initializer_list<T> il) : prevector(il.begin(), il.end()) {}
```
and the line here beomes
```C++
static prevector<ADDR_IPV6_SIZE, uint8_t> ipv6_loopback{0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1};
```

Reply:
I wonder if we can avoid defining a local variable for IPv6 loopback address and use `IN6ADDR_LOOPBACK_INIT` or `in6addr_loopback`...